### PR TITLE
feat: release on luarocks

### LINKED ISSUE #1: Publish it to luarocks for rocks.nvim?
https://github.com/nvim-neorocks/rocks.nvim can allow user don't declare dependence in package manager.
How about support it?
TIA!

### PR DESCRIPTION
**Pull request by Freed-Wu**
Fix #1
Refer https://github.com/mg979/vim-visual-multi/pull/262

### CODE REVIEW

**Comment by black-desk** [comment]
I don't think it is right time for this plugin to be released to non-developer user.

It has so many bugs that I meet all the day. I just do not have time to fix them.

I take a look on rocks.nvim and I think you can just use https://github.com/nvim-neorocks/rocks-git.nvim to install this plugin.

**Comment by Freed-Wu** [comment]
Or can we publish a scm version? It should give a chance to those users want to adventure.

rocks-git doesn't support declare a dependence, user still declare by themselves.

**Comment by black-desk** [comment]
> Or can we publish a scm version? It should give a chance to those users want to adventure.

I am not familiar with luarocks, may be you mean something like the *-git package in AUR?

**Comment by black-desk** [comment]
And check your email inbox. I send some message to you before.

**Comment by Freed-Wu** [comment]
Here is scm: https://github.com/black-desk/fcitx5-ui.nvim/pull/8/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R21

Just like *-git pakcages in AUR. You are right.

**Comment by black-desk** [comment]
> rocks-git doesn't support declare a dependence, user still declare by themselves.

Are you sure about this?

https://github.com/search?q=repo%3Anvim-neorocks%2Frocks-git.nvim%20rockspec&type=code

It seems that we could place a rockspec.lua file into this repository and dependencies will be installed.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+"on":
```
Comment (black-desk) [inline .github/workflows/main.yml:2]: remove these quotes.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    paths-ignore:
```
Comment (black-desk) [inline .github/workflows/main.yml:4]: Should we just limit this workflow on "master"?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - "**.md"
+  pull_request:
```
Comment (black-desk) [inline .github/workflows/main.yml:6]: Why run on pull_request?

**Comment by Freed-Wu** [comment]
If use rocks-git, it looks like:

```toml
[rocks]
dbus_proxy = "scm"

[plugins."fcitx5-ui.nvim"]
git = "black-desk/fcitx5-ui.nvim"
```

If we support luarocks directly, it is:

```toml
[plugins]
fcitx5-ui.nvim = "scm"
```

The latter should be better.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          LUAROCKS_API_KEY: ${{secrets.LUAROCKS_API_KEY}}
+        if: "! startsWith(github.ref, 'refs/tags/')"
+        with:
+          version: scm
+          dependencies: |
+            dbus_proxy
+      - uses: nvim-neorocks/luarocks-tag-release@v5
+        env:
+          LUAROCKS_API_KEY: ${{secrets.LUAROCKS_API_KEY}}
+        if: startsWith(github.ref, 'refs/tags/')
```
Comment (black-desk) [inline .github/workflows/main.yml:27]: I think we should not write workflow on tags right now as github action is hard to debug if you do not run them, and this project is not ready for release yet.

**Review by black-desk — APPROVED** [review]

**Review by black-desk — CHANGES_REQUESTED** [review]
opps, I means to request changes.

**Comment by black-desk** [comment]
And this commit should not be 'feat:' but 'ci:'.

Check https://www.conventionalcommits.org/en/v1.0.0/

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (black-desk) [inline .github/workflows/main.yml:1]: remove this line as well

**Comment by Freed-Wu** [comment]
Fixed.

**Pull request merged**